packages/dev/core/src/Lights/light.ts

```diff
@@ -485,10 +486,23 @@ export abstract class Light extends Node implements ISortableLight {
 
     /**
      * Returns the Light associated shadow generator if any.
+     * @param camera Camera for which the shadow generator should be retrieved (default: null). If null, retrieves the default shadow generator
      * @returns the associated shadow generator.
      */
-    public getShadowGenerator(): Nullable<IShadowGenerator> {
-        return this._shadowGenerator;
+    public getShadowGenerator(camera: Nullable<Camera> = null): Nullable<IShadowGenerator> {
+        if (this._shadowGenerators === null) {
+            return null;
+        }
+
+        return this._shadowGenerators.get(camera) ?? null;

```

Comment:
Is this not a breaking change as if we pass nothing we might not get the shadow generator as it was before ???

Reply:
No, it is not a breaking change because existing code will call the shadow generator constructor with a `null` camera, so the generator will be stored in `light._shadowGenerators` with a null key. Then, in `getShadowGenerator` (the code you pointed to), camera will be `null` for existing code because the camera parameter did not exist. So it will retrieve the right generator, the one associated to `null`.

Reply:
Nailed it !!! Thanks